src/recommendations/RecommendationsList.jsx

```diff
@@ -15,6 +16,17 @@ const RecommendationsList = (props) => {
     title, recommendations, userId, setSelectedRecommendationsType, selectedRecommendationsType,
   } = props;
 
+  useEffect(() => {
+    if (recommendations.length) {
+      const viewedProductsList = recommendations.map((product) => ({
```

Comment:
this code should be part of the `trackRecommendationsViewed` function. We should simply pass products here.

Reply:
Addressed.